test/http/request.cpp

```diff
@@ -3,83 +3,204 @@
 // information on the copying conditions.
 
 #define CATCH_CONFIG_MAIN
-#include "src/ext/Catch/single_include/catch.hpp"
 
+#include "src/common/check_connectivity.hpp"
+#include "src/ext/Catch/single_include/catch.hpp"
+#include "src/http/request_impl.hpp"
 #include <measurement_kit/common.hpp>
+#include <measurement_kit/ext.hpp>
 #include <measurement_kit/http.hpp>
-
-#include "src/common/check_connectivity.hpp"
-#include "src/http/request.hpp"
+#include <openssl/md5.h>
 
 using namespace mk;
 using namespace mk::net;
 using namespace mk::http;
 
-// TODO: after refactoring there are probably duplicated tests, we
-// should merge them to avoid testing things twice
+// Either tor was running and hence everything should be OK, or tor was
+// not running and hence connect() to socks port must have failed.
+static inline bool check_error_after_tor(Error e) {
+    return e == NoError() or e == ConnectFailedError();
+}
```

Comment:
Just moved from below. This, for now, allows to cover the cases where Tor is running and where Tor instead is not running, but (note to self!) it would probably better to have a specific SOCKS5 err.